src/components/container/ConnectDevicesMenu/ConnectDevicesMenu.tsx

```diff
@@ -89,16 +98,18 @@ function useConnectDevicesMenuState(
         Promise.all([
             MyDataHelps.getDataCollectionSettings(),
             MyDataHelps.getExternalAccounts(),
-            MyDataHelps.getDeviceInfo()
+            MyDataHelps.getDeviceInfo(),
+            MyDataHelps.getParticipantInfo()
         ])
-            .then(([settings, accounts, deviceInfo]) => {
+            .then(([settings, accounts, deviceInfo, participantInfo]) => {
                 let newState: ConnectDevicesMenuState = {
                     loading: true,
```

Comment:
I know this existed before your changes, but it looks like this could be set to `false` here and the few `newState.loading = false;` calls below could be removed.  Since this `newState` never gets set as THE `state` while the flag is `true`.